kore/src/Kore/Parser/LexerWrapper.hs

```diff
@@ -0,0 +1,166 @@
+) where
+
+-- -----------------------------------------------------------------------------
+-- Alex wrapper code.
+--
+-- Taken and modified from public domain code that is part of Alex:
+-- https://github.com/haskell/alex/blob/3.2.6/templates/wrappers.hs
+import Control.Applicative as App (Applicative (..))
+import qualified Data.ByteString as ByteString
+import qualified Data.ByteString.Internal as ByteString hiding (ByteString)
+import qualified Data.ByteString.Unsafe as ByteString
+import qualified Data.Char
+import Data.Int
+import Data.Word (Word8)
+import Kore.Parser.Lexer
+import Prelude.Kore
+
+type Byte = Word8
+
+-- -----------------------------------------------------------------------------
+-- The input type
+
+ignorePendingBytes :: AlexInput -> AlexInput
+ignorePendingBytes i = i -- no pending bytes when lexing bytestrings
+
+alexInputPrevChar :: AlexInput -> Char
+alexInputPrevChar = alexChar
+
+-- -----------------------------------------------------------------------------
+-- Token positions
+
+-- `Posn' records the location of a token in the input text.  It has three
+-- fields: the address (number of characters preceding the token), line number
+-- and column of a token within the file. `start_pos' gives the position of the
+-- start of the file and `eof_pos' a standard encoding for the end of file.
+-- `move_pos' calculates the new position after traversing a given character,
+-- assuming the usual eight character tab stops.
+
+alexStartPos :: FilePath -> AlexPosn
+alexStartPos fp = AlexPn fp 0 1 1
+
+-- -----------------------------------------------------------------------------
+-- Monad (default and with ByteString input)
+
+-- Compile with -funbox-strict-fields for best results!
+
+runAlex :: FilePath -> ByteString.ByteString -> Alex a -> Either String a
+runAlex fp input__ (Alex f) =
+    case f
+        ( AlexState
+            { alex_bpos = 0
+            , alex_pos = alexStartPos fp
+            , alex_inp = input__
+            , alex_chr = '\n'
+            , alex_scd = 0
+            }
+        ) of
+        Left msg -> Left msg
+        Right (_, a) -> Right a
+alexGetInput :: Alex AlexInput
+alexGetInput =
+    Alex $ \s@AlexState{alex_pos = pos, alex_bpos = bpos, alex_chr = c, alex_inp = inp__} ->
+        Right (s, AlexInput{alexPosn = pos, alexChar = c, alexStr = inp__, alexBytePos = bpos})
+
+alexSetInput :: AlexInput -> Alex ()
+alexSetInput AlexInput{alexPosn = pos, alexChar = c, alexStr = inp__, alexBytePos = bpos} =
+    Alex $ \s -> case s
+        { alex_pos = pos
+        , alex_bpos = bpos
+        , alex_chr = c
+        , alex_inp = inp__
+        } of
+        state__@(AlexState{}) -> Right (state__, ())
+
+alexGetStartCode :: Alex Int
+alexGetStartCode = Alex $ \s@AlexState{alex_scd = sc} -> Right (s, sc)
+
+alexSetStartCode :: Int -> Alex ()
+alexSetStartCode sc = Alex $ \s -> Right (s{alex_scd = sc}, ())
+
+alexEOF :: Alex Token
+alexEOF = do
+    AlexInput{alexPosn = p} <- alexGetInput
```

Comment:
These should probably all be `INLINE`'d. Not sure how much difference that will make if any.
(Github is being a bit weird about the region I highlighted: I am referring to `alexGetInput` through `alexEOF`.)